lib/MetaProcessor/MetaParser.cpp

```diff
@@ -474,6 +474,34 @@ namespace cling {
     return false;
   }
 
+  // dumps/creates a trace of the requested representation.
+  bool MetaParser::istraceCommand() {
+    if (getCurTok().is(tok::ident) &&
+        getCurTok().getIdent().equals("trace")) {
+      consumeToken();
+      skipWhitespace();
+      if (!getCurTok().is(tok::ident))
+          return false;
+      std::string ident = getCurTok().getIdent();
+      consumeToken();
+      skipWhitespace();
+      if (getCurTok().is(tok::eof)) {
+        m_Actions->actOnstatsCommand(ident=="ast" 
+            ? llvm::StringRef("asttree") : llvm::StringRef(ident), llvm::StringRef());
```

Comment:
The useless conversion from llvm::StringRef to std::string and back to llvm::StringRef is more important to me.

The other thought was that Interpreter::dump understands **asttree** to match the syntax you originally proposed. If that has changed because **asttree** is an improper noun then Interpreter::dump should not be using it either. The two solutions I see are to change **asttree** to **astree** or:

```
m_Interp.dump("ast", "") -> equivalent to .stats ast
m_Interp.dump("ast", "*") -> equivalent to .trace ast
m_Interp.dump("ast", "name") -> equivalent to .trace ast name
```

But that can also be handled in a subsequent PR...